models/prices/bitcoin/prices_bitcoin_tokens.sql

```diff
@@ -0,0 +1,30 @@
+{{ config(
+        schema='prices_bitcoin',
+        alias = 'tokens',
+        materialized='table',
+        file_format = 'delta',
+        tags = ['static']
+        )
+}}
+SELECT 
+    token_id
+    , blockchain
+    , symbol
+    , contract_address
```

Comment:
if you plan to union this into `prices_tokens`, which it appears you are, then this field will need casted to varbinary to match (hence the current CI failure in the gh workflow).
`from_base58(contract_address) as contract_address`
similar to the solana one [here](https://github.com/duneanalytics/spellbook/blob/main/models/prices/solana/prices_solana_tokens.sql#L13). the downstream implication means you'll need to cast it back to varchar on queries in order to join to bitcoin base tables

Reply:
@jeff-dude 
There are few complications here. Let me elaborate. 
------------------
Bitcoin:
Bitcoin doesn't have smart contracts. So we can't really talk about contract_address, what is represented here is not even a bitcoin address but is rather known as 'inscription id'. As an example , here are two explorers showing info about 'SATS' ordinal:

https://ordinalswallet.com/inscription/9b664bdd6f5ed80d8d88957b63364c41f3ad4efb8eee11366aa16435974d9333i0

https://ordiscan.com/inscription/357097

The owner (deployer) of SATS is the bitcoin address 'bc1prtawdt82wfgrujx6d0heu0smxt4yykq440t447wan88csf3mc7csm3ulcn' but this address could be the deployer of many other inscriptions. Therefore, it can't uniquely represent SATS. The best way of representing SATS is by using its inscription ID '9b664bdd6f5ed80d8d88957b63364c41f3ad4efb8eee11366aa16435974d9333i0'. But this id is not a base_58 address, so using from_base58 would fail. 

I'm really not an expert here but I see two possibilities:
1/ Similar to prices_native_tokens spell, we could use null as value for contract_address. This is my preferred solution as these are not contract addresses, and we are only interested in the price of the ticker anyway.

2/ The only from_base.. Dune function  that doesn't fail when using the inscription IDs is from_base64. Although this is not a base 64, it's the only non-breaking available function that could be used. If strictly required to have something, we could use from_base_64 of either the inscrption id or the inscription number (ex: 357097 for SATS). I personally don't like this solution since it's confusing. 

------------------
Cardano:
Cardano uses Base58 for Byron addresses and [Bech32](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) for Shelley. Bech32 is at most 90 characters long . 
Source: 
https://cips.cardano.org/cip/CIP-0019

Solutions for cardano addresses are similar to bitcoin but with also a third option:
1/ Dune implementing a from_bech32 function 

---------
Given the scope of this spell (getting the price for a ticker), and the complications and nuances brought by current and possible future non-evm standards, it seems to me that just using a null value for the contract address (similar to prices_native_tokens) seems like the easiest option. 
I'm adding a tentative commit accordingly but please let me know if it needs to be updated or we need to take a different approach.

Reply:
got it, thanks for the context.

i agree, forcing null address on both works for me. you will need to update each schema file test to be like the native one:
https://github.com/duneanalytics/spellbook/blob/main/models/prices/prices_schema.yml#L38-L42